### PR TITLE
PassManager.passes() returns the passes that had been added to the pass manager

### PR DESCRIPTION
**Pull request by 1ucian0**
This PR includes https://github.com/Qiskit/qiskit-terra/pull/2019

This new `PassManager.passes()` method returns the passes that had been added to the pass manager.

### CODE REVIEW

**Comment by ajavadia** [comment]
Can we have the most common case (i.e. a linear list of passes) just return that (a list of passes). This is very general but can be hard to parse in simple/common cases.

**Comment by 1ucian0** [comment]
I would like to keep the generality. Would a pretty printing function help?
